### PR TITLE
reimplement get_commensurate_diagonal_supercell

### PR DESCRIPTION
**Pull request by xivh**
no tests

### CODE REVIEW

File: src/spgrep_modulation/utils.py [hunk 1]
```diff
@@ -72,15 +73,19 @@ def qr_unique(a: NDArrayComplex) -> tuple[NDArrayComplex, NDArrayComplex]:
     return q, r
 
 
-def get_commensurate_diagonal_supercell(qpoint: NDArrayFloat, atol: float = 1e-8) -> NDArrayInt:
-    """Return minimum diagonal supercell in which ``qpoint`` is commensurate."""
+def get_commensurate_diagonal_supercell(qpoint: NDArrayFloat, max_size: int = 10) -> NDArrayInt:
```
Comment (lan496) [inline src/spgrep_modulation/utils.py:76]: Can you change `max_size` to `max_denominator` here and docstring?
That is to be consistent with other functions for example: https://github.com/phonopy/spgrep-modulation/blob/c347ab984890ad28f9777a0f11f240c962e4e422/src/spgrep_modulation/isotropy.py#L23-L34

**Comment by lan496** [comment]
Thanks, @xivh.
I've commented nitpicking to your PR, but the other parts look good to me!

**Comment by xivh** [comment]
I made that change and added some tests. The test checks if the supercell matrix times the qpoint contains all integers, but it fails for [0.59, 0.61, 0.63].

Do you think that there should be an additional tolerance parameter that gives a either a warning or raises an error if max_denominator is too small?

**Comment by lan496** [comment]
@xivh Thank you for handling my comment!

> Do you think that there should be an additional tolerance parameter that gives a either a warning or raises an error if max_denominator is too small?

I think a warning is enough. So, adding some `atol` and checking a supercell matrix will be more kind like below.

```python
supercell = get_commensurate_diagonal_supercell(qpoint)
qpoint_in_supercell = supercell @ qpoint
qpoint_in_suercell -= np.rint(qpoint_in_suercell)
if not np.allclose(qpoint_in_suercell, 0, atol=atol):
  # warn here
```

Also, [0.59, 0.61, 0.63] and [0.33, 0.33, 0.33] can be omitted from testcases for now.

**Comment by xivh** [comment]
I added a tolerance plus warning, but with the default parameters I was not able to trigger the warning.

**Comment by lan496** [comment]
@xivh LGTM!
Is it ready to be merged? If so, I'll merge it.

**Comment by xivh** [comment]
Yes, I think so.

**Pull request merged**